File: 8/10/3.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;


bool dfs(int start, vector<vector<int>>& computers){
   
    if(computers[start][start] == 0)
        return false;

    computers[start][start] = 0;
    for(int i = 0; i < computers.size(); i++ ){
        if(computers[start][i] == 1){
            dfs(i,computers);
        }



    }


    return true;

    




}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    

    for(int i = 0; i <n ; i++){

        if(computers[i][i] == 1 && dfs(i,computers)){
            answer++;
        }


    }



    return answer;
}

File: 2021/2/12/3.py
#프로그래머스 하노이의 탑
answer = []

def move(n,start,end) :
    answer.append([start,end])

def hanoi(n,start,end,via) :
    if n == 1 :
        move(n,start,end)
    else :
        hanoi(n-1,start,via,end)
        move(n,start,end)
        hanoi(n-1,via,end,start)


def solution(n):

    hanoi(n,1,3,2)
    return answer

print(solution(2))